### PR TITLE
kubeadm: switch jobs to eks cluster; add test job for encryption-algorithm

### PR DESCRIPTION
**Pull request by neolit123**
xref https://github.com/kubernetes/kubeadm/pull/3077
xref https://github.com/kubernetes/kubeadm/issues/3073

### CODE REVIEW

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-encryption-algorithm.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# AUTOGENERATED by https://git.k8s.io/kubeadm/kinder/ci/tools/update-workflows
+periodics:
+
+- name: ci-kubernetes-e2e-kubeadm-kinder-encryption-algorithm-latest
+  cluster: k8s-infra-prow-build
```
Comment (BenTheElder) [inline config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-encryption-algorithm.yaml:5]: todo: later, can we use the eks cluster for some of these?

Comment (neolit123) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-encryption-algorithm.yaml:5]: i though the jobs were already migrated.
adding the update as part of this PR in a commit the generator is updated in the k/kubeadm PR too.

Comment (BenTheElder) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-encryption-algorithm.yaml:5]: they are migrated already, it's not high priority, but we want to load balance some to the AWS budget where we can, versus a lot of the "real" cluster e2e has to run on the GCP budget both for the test cluster and for the prow pod.

**Review by BenTheElder — COMMENTED** [review]
/lgtm
/approve
/hold

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-encryption-algorithm.yaml (hunk 1 above)
Comment (neolit123) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-encryption-algorithm.yaml]: i forgot that k8s-infra-prow-build is already community infra.
updated the PR again. we can try all kubeadm jobs on EKS for a while and see how it goes.

**Review by BenTheElder — COMMENTED** [review]
/lgtm

**Pull request merged**